.github/workflows/deploy.yml

```
@@ -0,0 +1,52 @@
+name: Deploy to GitHub pages
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

You are not using the latest actions.

```suggestion
      - uses: actions/checkout@v4
```